yt_dlp/extractor/floatplane.py

```diff
@@ -0,0 +1,137 @@
+    _TESTS = [{
+        'url': 'https://www.floatplane.com/post/DdShecU983',
+        'info_dict': {
+            'id': 'DdShecU983',
+            'ext': 'mp4',
+            'title': 'TQ: Are Intel Arc Graphics A Bad Idea?',
+            'thumbnail': r're:^https?://.*\.jpg$',
+        },
+        'skip': 'Requires premium Floatplane account',
+    }]
+
+    def _real_initialize(self):
+        self._download_webpage(
+            'https://www.floatplane.com/', None,
+            note='Fetching session cookie')
+        self.session_id = self._get_cookies(
+            'https://www.floatplane.com')['sails.sid'].value
+
+    def _real_extract(self, url):
+        post_id = self._match_id(url)
+
+        headers = {'Cookie': f"sails.sid={self.session_id}"}
+
+        ###
+        # Fetch post details
+        ###
+        post_api_url = f"https://www.floatplane.com/api/v3/content/post?id={post_id}"
+
+        post_metadata = self._download_json(
+            post_api_url, post_id, headers=headers, note='Fetching post details')
+
+        if not traverse_obj(post_metadata, ('metadata', 'hasVideo')):
+            raise UnsupportedError(
+                'Provided Floatplane post does not contain a video')
+
+        ###
+        # Fetch video details
+        ###
+        video_id = traverse_obj(post_metadata, ('videoAttachments', 0, 'id'))
+        video_api_url = f"https://www.floatplane.com/api/v3/content/video?id={video_id}"
+
+        video_metadata = self._download_json(
+            video_api_url, post_id, headers=headers, note='Fetching video details')
+
+        ###
+        # Fetch video format details
+        ###
+        video_format_api_url = f"https://www.floatplane.com/api/v2/cdn/delivery?type=vod&guid={video_metadata['guid']}"
+
+        video_format_metadata = self._download_json(
+            video_format_api_url, post_id, headers=headers, note='Fetching video format details')
+
+        # Generate formats
+        remote_qualities = traverse_obj(
+            video_format_metadata, ('resource', 'data', 'qualityLevels'))
+        remote_qualities_params = traverse_obj(
+            video_format_metadata, ('resource', 'data', 'qualityLevelParams'))
+        remote_cdn = video_format_metadata['cdn']
+        format_path_template = traverse_obj(
+            video_format_metadata, ('resource', 'uri'))
+        formats = []
+        for currentFormat in remote_qualities:
+            currentParams = remote_qualities_params[currentFormat['name']]
+            extension = currentParams["2"].split(
+                '.')[1]  # Get 'mp4' from '720p.mp4'
+
+            replaced_path = format_path_template.replace('{qualityLevelParams.2}', currentParams['2']).replace(
+                '{qualityLevelParams.4}', currentParams['4'])
+
+            formats.append({
+                'format_id': currentFormat['name'],
+                'width': currentFormat['width'],
+                'height': currentFormat['height'],
```

Comment:
make non-fatal. Same below. Read https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#mandatory-and-optional-metafields